File: src/main/java/com/study/platform/domain/post/event/PostSyncEvent.java
package com.study.platform.domain.post.event;

import io.swagger.v3.oas.annotations.media.Schema;

import java.util.UUID;

@Schema(description = "ES 동기화 이벤트")
public record PostSyncEvent(
        @Schema(description = "게시글 ID")
        UUID postId,

        @Schema(description = "ES 동기화 작업 타입")
        PostSyncOperationType operationType,

        @Schema(description = "Outbox 이벤트 ID")
        Long outboxEventId,

        int retryCount
) {
    public PostSyncEvent withRetry() {
        return new PostSyncEvent(postId, operationType, outboxEventId, retryCount + 1);
    }
}

File: domain/src/main/kotlin/com/study/platform/domain/post/event/PostSyncEvent.kt
package com.study.platform.domain.post.event

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import java.util.UUID

data class PostSyncEvent @JsonCreator constructor(
    @param:JsonProperty("postId") val postId: UUID,
    @param:JsonProperty("operationType") val operationType: PostSyncOperationType,
    @param:JsonProperty("outboxEventId") val outboxEventId: Long,
    @param:JsonProperty("retryCount") val retryCount: Int
) {
    fun withRetry(): PostSyncEvent = copy(retryCount = retryCount + 1)
}
